apps/server/lib/lexical/server/project/diagnostics.ex

```diff
@@ -137,7 +138,19 @@ defmodule Lexical.Server.Project.Diagnostics do
     defp position_to_range(%SourceFile{} = source_file, {line_number, column}) do
       line_number = Math.clamp(line_number, 1, SourceFile.size(source_file))
       column = max(column, 1)
+      to_lsp_range(line_number, column, source_file)
+    end
+
+    defp position_to_range(source_file, line_number) when is_integer(line_number) do
+      line_number = Math.clamp(line_number, 1, SourceFile.size(source_file))
+
+      with {:ok, line_text} <- SourceFile.fetch_text_at(source_file, line_number) do
+        column = Text.count_leading_spaces(line_text) + 1
+        to_lsp_range(line_number, column, source_file)
+      end
+    end
 
+    defp to_lsp_range(line_number, column, source_file) do
```

Comment:
what happens if we don't make this an LSP range, and instead use a Lexical range?

Reply:
#103